hera_qm/vis_metrics.py

```diff
@@ -29,3 +30,173 @@ def check_noise_variance(data):
+            # get data and weights
+            d1 = uvd1.get_data(bl1)
+            w1 = (~uvd1.get_flags(bl1)).astype(np.float)
+            d2 = uvd2.get_data(bl2)
+            w2 = (~uvd2.get_flags(bl2)).astype(np.float)
+
+            # skip if completely flagged
+            if np.isclose(w1, 0.0).all() or np.isclose(w2, 0.0).all():
+                continue
+
+            # get means
+            m1 = np.sum(d1 * w1, axis=sumaxes, keepdims=True) / np.sum(w1, axis=sumaxes, keepdims=True).clip(1e-10, np.inf)
+            m2 = np.sum(d2 * w2, axis=sumaxes, keepdims=True) / np.sum(w2, axis=sumaxes, keepdims=True).clip(1e-10, np.inf)
+
+            # get cov
+            c = np.sum((d1 - m1) * w1 * (d2 - m2) * w2, axis=sumaxes, keepdims=True) / np.sum(w1 * w2).clip(1e-10, np.inf)
+
+            # assign
+            cov[i, j] = c
+
+    return cov
+
+
+def plot_bl_cov(uvd1, uvd2, bls, ax=None, cmap='viridis', vmin=None, vmax=None,
+                component='abs', colorbar=True, tlsize=10, tlrot=35, figsize=None):
+    """
+    Plot the visibility data covariance between uvd1 and uvd2
+    across a set of specified baselines.
+
+    Parameters
+    ----------
+    uvd1 : UVData object
+
+    uvd2 : UVData object
+
+    bls : list
+        List of baseline antenna-pairs
+
+    ax : matplotlib.axes.Axis object
+
+    cmap : str
+        Colormap to use
+
+    vmin, vmax : float
+        Colorscale min and max
+
+    component : str
+        Component of complex covariance to plot. Options = ['real', 'imag', 'abs'].
+
+    colorbar : bool
+        If True, plot a colorbar
+
+    tlsize : int
+        Tick-label size.
+
+    tlrot : int
+        Tick-label rotation in degrees.
+
+    figsize : tuple
+        Len-2 integer tuple for figure-size if ax is None.
+
+    Returns
+    -------
+    if ax is None:
+        fig : matplotlib.pyplot.Figure object
+    """
+    # get covariance
+    blcov = vis_bl_cov(uvd1, uvd2, bls).squeeze()
+    assert not np.isnan(blcov).all(), "All data are flagged!"
+    if component == 'abs':
+        blcov = np.abs(blcov)
+    elif component == 'real':
+        blcov = np.real(blcov)
+    elif component == 'imag':
+        blcov = np.imag(blcov)
+
+    # setup figure
+    if ax is None:
+        fig, ax = plt.subplots(figsize=figsize)
+        newfig = True
+    else:
+        fig = ax.get_figure()
+        newfig = False
+
+    # plot 
+    cax = ax.matshow(blcov, vmin=vmin, vmax=vmax, cmap=cmap, aspect='auto')
+
+    # ticks
+    Nbls = len(bls)
+    ax.xaxis.set_ticks_position('bottom')
+    ax.tick_params(labelsize=tlsize, rotation=tlrot)
+    ax.set_xticks(np.arange(Nbls))
+    ax.set_yticks(np.arange(Nbls))
+    ax.set_xticklabels(bls)
+    ax.set_yticklabels(bls)
+
+    # colorbar
+    if colorbar:
+        cbar = fig.colorbar(cax, ax=ax)
+        cbar.set_label(r"$\rm {}\ cov(V_{{1}},\ V_{{2}}^{{\ast}})\ [{}\ \cdot\ {}]$".format(component, uvd1.vis_units, uvd2.vis_units), fontsize=12)
```

Comment:
It's annoying sometimes with formatting strings but is it possible to make this line PEP8 compliant?

sometimes it can be broken at the  " mark:

    set_label("lots of formatting strings"
              .format(all the numbers))